lib/amazon_product_api.rb

```diff
@@ -0,0 +1,59 @@
+module AWSAPIs
+  class AmazonProductAPI
+    require 'httparty'
+    require 'time'
+    require 'uri'
+    require 'openssl'
+    require 'base64'
+
+    AWS_ACCESS_KEY = ENV["AWS_ACCESS_KEY"]
+    AWS_SECRET_KEY = ENV["AWS_SECRET_KEY"]
+    AWS_ASSOCIATES_TAG = ENV["AWS_ASSOCIATES_TAG"]
+
+    # The region you are interested in
+    ENDPOINT = "webservices.amazon.com"
+
+    REQUEST_URI = "/onca/xml"
+
+    def self.params
```

Comment:
Not important for this project but consider putting the static portions of the `params` `Hash` in a constant and then create a new `Hash` merging the constant part of the `params`.